app/models/registration.ts

```diff
@@ -44,15 +44,12 @@ export default class RegistrationModel extends NodeModel.extend() {
         'pendingEmbargoApproval', 'pendingEmbargoTerminationApproval',
         'pendingWithdrawal')
     get state(): RegistrationState {
-        return (
-            (this.pendingRegistrationApproval && RegistrationState.PendingRegistration) ||
-            (this.pendingEmbargoApproval && RegistrationState.PendingEmbargo) ||
-            (this.pendingEmbargoTerminationApproval && RegistrationState.PendingEmbargoTermination) ||
-            (this.pendingWithdrawal && RegistrationState.PendingWithdrawal) ||
-            (this.withdrawn && RegistrationState.Withdrawn) ||
-            (this.embargoed && RegistrationState.Embargoed) ||
-            RegistrationState.Public
-        );
+        const stateMap = this.registrationStateMap();
+        const currentState: any = Object.keys(stateMap)
```

Comment:
```suggestion
        const currentState: RegistrationState = Object.keys(stateMap)
```

Reply:
This can't be set to RegistrationState because inside the enum each value is a string - it complains that "RegistrationState is not a string"

Reply:
Ideas on this one? Only thing I couldn't implement

Reply:
I think you have to do:
```suggestion
        const currentState = (Object.keys(stateMap) as unknown as keyof typeof RegistrationState)
```
because `Object.keys()` *always* has a return type of `string[]`.

Reply:
Wait, my bad. This works:
```suggestion
        const currentState: any = (Object.keys(stateMap) as Array<keyof typeof RegistrationState>)
```

Reply:
Uuugh I keep forgetting how much I hate typescript enums. I find myself agreeing with [this article](https://medium.com/@katbusch/typescript-enums-explained-e5f9a101afc9) that we're probably better off using a union of [string literal types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types). 

but to make this work with an enum, something like this (building off @jamescdavis):
```ts
const currentState: RegistrationState = (Object.keys(b) as Array<keyof typeof RegistrationState>)
  .filter(k => c[k])
  .map(k => RegistrationState[k])
  [0];
```
or 
```ts
const currentState: RegistrationState = Object.keys(b)
  .filter(k => c[k])
  .map(k => RegistrationState[k as keyof typeof RegistrationState])
  [0];
```

Reply:
Yea this enum is also used as a sort of kind of interface thing. It's definitely not great. But I think the above bit works.

Reply:
+1 to string literal types.

Reply:
Done.